discovery-provider/src/tasks/entity_manager/entities/user.py

```diff
@@ -242,9 +257,13 @@ def update_user(params: ManageEntityParameters):
         params.web3,
         params.challenge_bus,
     )
-    user_record.metadata_multihash = params.metadata_cid
+
+    updated_metadata_cid = merge_user_metadata(params, user_record)
```

Comment:
shouldn't the merging happen before the validation steps? same for the other entities. otherwise how can you validate on the partial metadata?

Reply:
The validation null checks the fields it is validating, so it seems to work even with partial metadata